spam/03_spam_data_slicing_tutorial.py

```diff
@@ -0,0 +1,491 @@
+# -*- coding: utf-8 -*-
+# %% [markdown]
+# # ✂️ _Spam_ — Data Slicing Tutorial
+#
+# In real-world applications, some model outcomes are often more important than others — e.g. vulnerable cyclist detections in an autonomous driving task, or, in our running **spam** application, potentially malicious link redirects to external websites.
+#
+# Traditional machine learning systems optimize for overall quality, which may be too coarse-grained: models that achieve high overall performance might produce unacceptable failure rates on critical slices of the data — data subsets that might correspond to vulnerable cyclist detection in an autonomous driving task, or in our running spam detection application, external links to potentially malicious websites.
```

Comment:
Nit: these sentences are out of control. comma -> colon -> mdash -> comma -> comma. gets really hard to remember where i am by the end.

Reply:
split a sentence, good call